amazon/ion/simpleion.py

```diff
@@ -198,7 +198,7 @@ def _ion_type(obj, from_type):
     raise TypeError('Unknown scalar type %r' % (type(obj),))
 
 
-def _dump(obj, writer, from_type, field=None):
+def _dump(obj, writer, from_type, field=None, in_struct=False, depth=0):
```

Comment:
I noticed that events from release v0.4.0 provide the depth value; that's why I've re-added it here.  If we want to remove it, let's do so explicitly, rather than as a side-effect of #97.